compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/interop/UIKitInteropView.uikit.kt

```diff
@@ -0,0 +1,193 @@
+import androidx.compose.ui.layout.MeasurePolicy
+import androidx.compose.ui.layout.MeasureScope
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.IntRect
+import androidx.compose.ui.unit.round
+import kotlinx.atomicfu.atomic
+import platform.CoreGraphics.CGRectMake
+import platform.UIKit.UIColor
+import platform.UIKit.UIView
+import platform.UIKit.addSubview
+import platform.UIKit.backgroundColor
+import platform.UIKit.insertSubview
+import platform.UIKit.removeFromSuperview
+import platform.UIKit.setFrame
+import platform.darwin.dispatch_async
+import platform.darwin.dispatch_get_main_queue
+
+private val NoOpUpdate: UIView.() -> Unit = {}
+
+/**
+ * @param modifier The modifier to be applied to the layout. Size should be specified in modifier.
+ * Modifier may contains crop() modifier with different shapes.
+ * @param background A color of UIView background.
+ * @param update A callback to be invoked after the layout is inflated.
+ * @param dispose A callback that will be called when the view leaves the composition.
+ * @param factory The block creating the [UIView] to be composed.
+ */
+@Composable
+fun <T : UIView> UIKitInteropView(
+    modifier: Modifier,
+    background: Color = Color.White,
+    update: (T) -> Unit = NoOpUpdate,
+    dispose: (T) -> Unit = {},
+    factory: () -> T,
+) {
+    val componentInfo = remember { ComponentInfo<T>() }
+    val root = LocalLayerContainer.current
+    val density = LocalDensity.current.density
+    var rectInPixels by remember { mutableStateOf(IntRect(0, 0, 0, 0)) }
+    var localToWindowOffset: IntOffset by remember { mutableStateOf(IntOffset.Zero) }
+    Place(
+        modifier.onGloballyPositioned { childCoordinates ->
+            val coordinates = childCoordinates.parentCoordinates!!
+            localToWindowOffset = coordinates.localToWindow(Offset.Zero).round()
+            val newRectInPixels = IntRect(localToWindowOffset, coordinates.size)
+            if (rectInPixels != newRectInPixels) {
+                val rect = newRectInPixels / density
+                componentInfo.container.setFrame(rect.toCGRect())
+                if (rectInPixels.width != newRectInPixels.width || rectInPixels.height != newRectInPixels.height) {
+                    componentInfo.component.setFrame(
+                        CGRectMake(
+                            0.0,
+                            0.0,
+                            rect.width.toDouble(),
+                            rect.height.toDouble()
+                        )
+                    )
+                }
+                rectInPixels = newRectInPixels
+            }
+        }.drawBehind {
+            drawRect(Color.Transparent, blendMode = BlendMode.DstAtop)//draw transparent hole
+        }
+    )
+
+    DisposableEffect(factory) {
```

Comment:
add update?

Reply:
Checked.
update value changes also in SideEffect {
So we don't need this as a key in DisposableEffect()